src/luxonis_ml/data/dataset.py

```diff
@@ -313,6 +320,12 @@ def _end_time(self) -> None:
         self.t1 = time.time()
         print(f"Took {self.t1 - self.t0} seconds")
 
+    def _make_temp_dir(self) -> None:
```

Comment:
Might be better to create a context manager for this. A little cleaner to read and also, we ensure the 
directory gets deleted in case some exception happens.

```python
@contextmanager
def _temp_dir(self):
    os.makedirs(self.tmp_dir, exist_ok=True)
    yield
    shutil.rmtree(self.tmp_dir)
```

Reply:
Yeah, I like the idea but I don't think it would work well with the other cases. I've added a line at initialization of `_make_temp_dir` to ensure the directory is removed to begin with in case there are some leftover files, so it would at least prevent cases where an exception causes leftover files to be wrongfully synced.